File: api/app/schemas/category.py
from datetime import datetime
from pydantic import BaseModel, Field

class CategoryCreateRequest(BaseModel):
    name: str = Field(min_length=1, max_length=80)
    kind: str = Field(pattern="^(income|expense|transfer|credit_payment)$")
    color: str | None = Field(default=None, max_length=20)
    icon: str | None = Field(default=None, max_length=50)
    is_system: bool = False

class CategoryUpdateRequest(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=80)
    kind: str | None = Field(default=None, pattern="^(income|expense|transfer|credit_payment)$")
    color: str | None = Field(default=None, max_length=20)
    icon: str | None = Field(default=None, max_length=50)

class CategoryResponse(BaseModel):
    id: str
    name: str
    kind: str
    color: str | None = None
    icon: str | None = None
    is_system: bool
    created_at: datetime
    updated_at: datetime

File: mobile/lib/features/categories/models/category.dart
enum CategoryKind { income, expense, transfer, creditPayment }

class Category {
  const Category({
    required this.id,
    required this.name,
    required this.kind,
    this.color,
    this.icon,
    this.isSystem = false,
  });

  final String id;
  final String name;
  final CategoryKind kind;
  final String? color;
  final String? icon;
  final bool isSystem;

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      id: json['id'] as String,
      name: json['name'] as String,
      kind: _parseKind(json['kind'] as String),
      color: json['color'] as String?,
      icon: json['icon'] as String?,
      isSystem: json['is_system'] as bool? ?? false,
    );
  }

  static CategoryKind _parseKind(String kind) {
    switch (kind.toLowerCase()) {
      case 'income':
        return CategoryKind.income;
      case 'expense':
        return CategoryKind.expense;
      case 'transfer':
        return CategoryKind.transfer;
      case 'credit_payment':
        return CategoryKind.creditPayment;
      default:
        return CategoryKind.expense;
    }
  }
}
